cpp/ql/lib/semmle/code/cpp/ir/implementation/raw/internal/TranslatedElement.qll

```diff
@@ -865,6 +870,17 @@ abstract class TranslatedElement extends TTranslatedElement {
       1 + sum(TranslatedElement child | child = this.getChildByRank(_) | child.getDescendantCount())
   }
 
+  /**
+   * Holds if this element has implicit destructor calls that should follow it.
+   */
+  predicate hasImplicitDestructorCalls() { none() }
+
+  /**
+   */
+  int getFirstDestructorCallIndex() { none() }
+
+  predicate handlesDestructorsExplicitly() { none() }
```

Comment:
QLDoc here would be nice. In particular: What does it mean to "handle" destructors explicitly?